### PR TITLE
dstor: fix PMIx_Get

### PR DESCRIPTION
**Pull request by karasevb**
- Fix get some data from the internal hash in dstor case.
- Silence a warning

### CODE REVIEW

File: src/client/pmix_client_get.c [hunk 1]
```diff
@@ -564,23 +564,24 @@ static void _getnbfn(int fd, short flags, void *cbdata)
...
+        cb->value_cbfunc(rc, val, cb->cbdata);
+        /* cleanup */
+        if (NULL != val) {
+            PMIX_VALUE_RELEASE(val);
+        }
+        PMIX_RELEASE(cb);
+        return;
     }
 
     /* otherwise, the data must be something they "put" */
```
Comment (artpol84) [inline src/client/pmix_client_get.c]: As far as I recall the the key name that caused particular error for OMPI was "pmix.loc". But what if the key with some arbitrary name was submitted through PMIx_Store_local?

File: src/common/pmix_jobdata.c [hunk 2]
```diff
@@ -136,6 +136,9 @@ static inline pmix_status_t _job_data_store(const char *nspace, void *cbdata)
     int32_t cnt;
     size_t nnodes;
     uint32_t i;
+#if !(defined(PMIX_ENABLE_DSTORE) && (PMIX_ENABLE_DSTORE == 1))
+    uint32_t j;
+#endif
```
Comment (artpol84) [inline src/common/pmix_jobdata.c]: Why this variable was introduced but no usages are visible?

File: src/client/pmix_client_get.c (hunk 1 above)
Comment (artpol84) [inline src/client/pmix_client_get.c]: I mean that we probably need to search "internal" hash in case key doesn't begin's from "pmix" too.

**Comment by artpol84** [comment]
:)

**Comment by artpol84** [comment]
@rhc54 - my sister sent it to me today, so I was unable to resist - it was fitting to well ;)

**Comment by karasevb** [comment]
bot:retest

**Comment by artpol84** [comment]
@karasevb :+1:

**Comment by jladd-mlnx** [comment]
😳

**Pull request merged**